src/LayerEditor/widgets/text_validator.py

```diff
@@ -0,0 +1,25 @@
+from PyQt5.QtGui import QValidator
+
+
+class TextValidator(QValidator):
```

Comment:
I was trying to make it so it could be used in other cases. But im not sure if it is such a good solution. What exception can occur? Default name is the original name before editing started. Maybe its not the best name.

Reply:
- Better documentation. validate and fixup are overridden methods of QValidator.
- Possibly i_text_unique_fnc replace by 'existing_names_fnc'.
- Rename ? to NameValidator